torchgeo/trainers/change.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for change detection."""
+
+import os
+from typing import Any
+
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+from einops import rearrange
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.classification import (
+    BinaryAccuracy,
+    BinaryF1Score,
+    BinaryJaccardIndex,
+)
+from torchvision.models._api import WeightsEnum
+
+from ..models import FCSiamConc, FCSiamDiff, get_weight
+from . import utils
+from .base import BaseTask
+
+
+class ChangeDetectionTask(BaseTask):
+    """Change Detection. Currently supports binary change between two timesteps."""
+
+    def __init__(
+        self,
+        model: str = 'unet',
+        backbone: str = 'resnet50',
+        weights: WeightsEnum | str | bool | None = None,
+        in_channels: int = 3,
+        pos_weight: Tensor | None = None,
+        loss: str = 'bce',
+        lr: float = 1e-3,
+        patience: int = 10,
+        freeze_backbone: bool = False,
+        freeze_decoder: bool = False,
+        threshold: float = 0.5,
```

Comment:
Technically we don't expose the threshold in `SemanticSegmentationTask(task='binary')`, but this is probably a good idea